### PR TITLE
Update to `FourierFlows.jl` v0.10.0

### PR DESCRIPTION
**Pull request by jbisits**
Update to `FourierFlows.jl` v0.10.0. This update has a breaking change with the device `dev` now being part of the `grid`. Need to redefine the `grid`s in the various methods for `Problem` and can update functions so that they find the `dev` through the `grid`. I also added a default argument for `dev`, `dev::Device=CPU()`, so that if this is not supplied `Problem` uses the `CPU()`.

### CODE REVIEW

**Comment by jbisits** [comment]
Need to update `GeophysicalFlows.jl` first as this is a dependency.

**Comment by navidcy** [comment]
Push here: https://github.com/FourierFlows/GeophysicalFlows.jl/pull/305

File: Project.toml [hunk 1]
```diff
@@ -20,8 +20,8 @@ Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 [compat]
 CUDA = "^1.3, ^2, ^3"
 DocStringExtensions = "0.8, 0.9"
-FourierFlows = "0.9.2"
-GeophysicalFlows = "0.14"
+FourierFlows = "0.10.0"
+GeophysicalFlows = "0.15"
```
Comment (navidcy) [inline Project.toml:24]:
```suggestion
GeophysicalFlows = "0.15.0”
```

not sure if it matters though

**Comment by navidcy** [comment]
@jbisits this is great. I pushed some fixes to a few things I could see. But when GeophysicalFlows.jl v0.15.0 is registered we should run the tests and see what happens.

**Review by navidcy — APPROVED** [review]
Also this is a breaking change to the package so we should bump up a minor release -- can you edit the `Project.toml`?

**Comment by navidcy** [comment]
- `@devzeros` takes a type of device, not the device
- I thought that `Vars` should also infer the type (e.g., Float64) from the grid rather than take it as kwarg.

**Comment by jbisits** [comment]
> * `@devzeros` takes a type of device, not the device
> * I thought that `Vars` should also infer the type (e.g., Float64) from the grid rather than take it as kwarg.

Can do!

**Comment by jbisits** [comment]
> Also this is a breaking change to the package so we should bump up a minor release -- can you edit the `Project.toml`?

Can do!

**Comment by jbisits** [comment]
Ah that is the minor release..!! My bad

**Comment by jbisits** [comment]
I am having some local issues with compiling. I get an error for `CUDA.jl` 

``julia
ERROR: LoadError: UndefVarError: QRCompactWY not defined
```

This is related to [this](https://github.com/JuliaGPU/CUDA.jl/pull/1558) but I have not solved what is happening yet.

**Comment by navidcy** [comment]
Hm.... There were some issues with `Problem(dev=CPU, advecting_flow; kwargs)`.... My compiling issues were coming from there. Try now?

**Comment by jbisits** [comment]
> Hm.... There were some issues with `Problem(dev=CPU, advecting_flow; kwargs)`.... My compiling issues were coming from there. Try now?

Mine was a `Manifest.toml` issue. Compiled now!

File: examples/cellularflow.jl [hunk 2]
```diff
@@ -37,7 +37,7 @@ nothing # hide
 
 # ## Numerical parameters and time-stepping parameters
 
-L = 2π        # domain size
+Lx = 2π        # domain size
```
Comment (navidcy) [inline examples/cellularflow.jl:40]:
```suggestion
Lx = 2π       # domain size
```

File: examples/cellularflow.jl [hunk 3]
```diff
@@ -62,7 +62,7 @@ nothing # hide
 
 # ## Problem setup
 # We initialize a `Problem` by providing a set of keyword arguments.
-prob = TracerAdvectionDiffusion.Problem(dev, advecting_flow; nx=n, Lx=L, κ, dt, stepper)
+prob = TracerAdvectionDiffusion.Problem(dev, advecting_flow; nx, Lx=Lx, κ, dt, stepper)
```
Comment (navidcy) [inline examples/cellularflow.jl:65]:
```suggestion
prob = TracerAdvectionDiffusion.Problem(dev, advecting_flow; nx, Lx, κ, dt, stepper)
```

File: examples/cellularflow.jl [hunk 4]
```diff
@@ -27,7 +27,7 @@ nothing # hide
 
 # ## Numerical parameters and time-stepping parameters
 
-      n = 128            # 2D resolution = n²
+     nx = 128            # 2D resolution = n²
```
Comment (navidcy) [inline examples/cellularflow.jl:30]:
```suggestion
     nx = 128            # 2D resolution = nx²
```

File: examples/cellularflow.jl (hunk 2 above)
Comment (navidcy) [inline examples/cellularflow.jl:40]:
```suggestion
Lx = 2π       # domain size
```

**Comment by navidcy** [comment]
I pushed a few minor changes to ensure things run smooth on GPU

**Comment by jbisits** [comment]
> I pushed a few minor changes to ensure things run smooth on GPU

Great!! I don't have a gpu so often always relying on tests to catch things.

**Comment by jbisits** [comment]
Was just writing to you about that!

**Comment by jbisits** [comment]
If this passes and looks all in order to you feel free to merge!

**Comment by navidcy** [comment]
Yeah, my bad.

**Pull request merged**